src/lib/draw-and-pick.js

```diff
@@ -136,16 +136,21 @@ export function pickLayers(gl, {
 
     affectedLayers.forEach(layer => {
       let info = Object.assign({}, baseInfo);
-      info.layer = layer;
 
       if (layer === pickedLayer) {
         info.color = pickedColor;
         info.index = pickedObjectIndex;
         info.picked = true;
       }
 
-      // Let layers populate its own info object
-      info = layer.pickLayer({info, mode});
+      // walk up the composite chain and find the owner of the event
+      // sublayers are never directly exposed to the user
+      while (layer && info) {
+        info.layer = layer;
+        // Let layers populate its own info object
+        info = layer.pickLayer({info, mode});
```

Comment:
Downside of a nested info object is that if the implementation of a layer is changed (for example, `PolygonLayer` went from a standalone layer to a composite layer), the structure of the picking info may also change. I like the simplicity of our current API where callbacks only need to look at `info.object` for the details of the picked object.